### PR TITLE
Content-Box flag and some fixes

### PR DESCRIPTION
**Pull request by pablomayobre**
Well I wanted to implement a content-box flag in order to support shadows and stuff.

When I did this I noticed paddings were inverted (if you wanted to add them you had to subtract them) so I fixed that.

Then I noticed that the values returned by `get_content_box` could differ from those returned by `draw`, for example if the width was less than `p.static.x` so I fixed that, and I did the same for `get_border_box`. But both functions computed the same stuff twice, and calling them in `draw` meant repeating stuff, so I enclosed that in a `preprocessed` flag.

When debugging I discovered that the red rectangle, that of the border box, was off, the width was different than that of the real box drawn by the patch, this was due to `p.static.x` being subtracted (same for the height), so instead of using `x, y, w, h` I changed it so that it uses `get_border_box(cx, cy, cw, ch)`

### CODE REVIEW

**Comment by pablomayobre** [comment]
PS: I already tested it, but any other test is welcomed. If you find any bugs or issues with this PR, tell me and I'll fix it

File: patchy.lua [hunk 1]
```diff
@@ -136,37 +136,64 @@ local function horizontal(state)
...
+	if not preprocessed then
+		x, y = (x or 0), (y or 0)
+		w = math.max((w or 0) - p.static.x, 0) + p.static.x
+		h = math.max((h or 0) - p.static.y, 0) + p.static.y
+	end
+
+	return x + p.pad[4], y + p.pad[1], w - p.pad[2] - p.pad[4] + p.static.x, h - p.pad[1] - p.pad[3] + p.static.y
+end
+-- Border Box (content + padding)
+local function get_border_box(p, x, y, w, h, preprocessed)
```
Comment (pablomayobre) [inline patchy.lua:150]: Maybe this should be:

``` lua
local function get_border_box(p, cx, cy, cw, ch, preprocessed)
    if not preprocessed then
        cx, cy = (x or 0), (y or 0)
        cw, ch = (w or 0), (h or 0)
    end

    local x, y, w, h = cx - p.pad[4], cy - p.pad[1], cw + p.pad[2] + p.pad[4], ch + p.pad[1] + p.pad[3]

    if not preprocessed then
        w = math.max(w - p.static.x, 0) + p.static.x
        h = math.max(h - p.static.y, 0) + p.static.x
    end
    return x, y, w, h
 end
```

**Comment by shakesoda** [comment]
Works for me, merged and bumped to version 1.2.0

**Comment by pablomayobre** [comment]
Nice! Thanks

**Pull request merged**